common/src/main/java/com/wynntils/models/gear/type/GearType.java

```diff
@@ -218,31 +209,23 @@ public boolean isValidWeapon(ClassType classType) {
     }
 
     private List<Float> getModelList() {
-        if (modelList == null) {
-            List<Float> tmp = new ArrayList<>();
-
-            if (minModelSupplier != null
-                    && maxModelSupplier != null
-                    && minModelSupplier.get().isPresent()
-                    && maxModelSupplier.get().isPresent()) {
-                int min = minModelSupplier.get().get().intValue();
-                int max = maxModelSupplier.get().get().intValue();
+        if (modelList.isEmpty()) {
+            List<Float> tempModelList = new ArrayList<>();
 
-                IntStream.rangeClosed(min, max).mapToObj(i -> (float) i).forEach(tmp::add);
-            }
-
-            if (minSkinModelSupplier != null
-                    && maxSkinModelSupplier != null
-                    && minSkinModelSupplier.get().isPresent()
-                    && maxSkinModelSupplier.get().isPresent()) {
-                int min = minSkinModelSupplier.get().get().intValue();
-                int max = maxSkinModelSupplier.get().get().intValue();
-
-                IntStream.rangeClosed(min, max).mapToObj(i -> (float) i).forEach(tmp::add);
-            }
+            addRangeToList(modelRange, tempModelList);
+            addRangeToList(skinModelRange, tempModelList);
 
-            modelList = List.copyOf(tmp);
+            modelList = List.copyOf(tempModelList);
         }
+
         return modelList;
     }
+
+    private static void addRangeToList(Supplier<Optional<Pair<Float, Float>>> supplier, List<Float> out) {
```

Comment:
Do we need to do this with suppliers instead of "burning in" the actual data at a suitable time (like construction)? In general, I don't like suppliers. The added flexibility can be great when really needed, but it makes it much harder to reason about the code and to analyze it in a static context. My understanding here is that this is just reflecting values from a json file we have read in, so there seems to be no need of this flexibility.